src/WhatToCook.WebApp/ClientApp/src/app/meal-planner/meal-plan-creator/meal-plan-creator.component.ts

```diff
@@ -1,81 +1,196 @@
+  name: FormControl<string>;
+  dates: FormGroup<MealPlanFormDates>;
+  recipes: FormArray<FormControl<string>>;
+}
 
 @Component({
   selector: 'app-meal-creator',
   templateUrl: './meal-plan-creator.component.html',
   styleUrls: ['./meal-plan-creator.component.scss'],
 })
 export class MealPlanCreatorComponent {
-  hasInteractedWithFromDate = false;
-  hasInteractedWithToDate = false;
-  mealPlanForm: FormGroup = new FormGroup(
-    {
-      name: this.fb.nonNullable.control('', [
-        Validators.required,
-        notWhitespaceValidator,
-      ]),
-      fromDate: this.fb.nonNullable.control(new Date(), [
-        Validators.required,
-        notPastDateValidator,
-      ]),
-      toDate: this.fb.nonNullable.control(new Date(), [
-        Validators.required,
-        notPastDateValidator,
-      ]),
-      recipes: this.fb.nonNullable.array([] as FormControl<string>[]),
-    },
-    { validators: dateRangeValidator },
-  );
-
-  handleSuccesfulSave() {
+  public mealPlans: MealPlanForDay[] = [];
+  public selectedMealPlanForDay?: MealPlanForDay;
+
+  mealPlanForm: FormGroup<MealPlanForm> = new FormGroup({
+    name: this.fb.nonNullable.control('', [
+      Validators.required,
+      notWhitespaceValidator,
+    ]),
+    dates: this.fb.nonNullable.group(
+      {
+        from: this.fb.control<string | null>(null, [
+          Validators.required,
+          notPastDateValidator,
+        ]),
+        to: this.fb.control<string | null>(null, [
+          Validators.required,
+          notPastDateValidator,
+        ]),
+      },
+      { validators: dateRangeValidator }
+    ),
+    recipes: this.fb.nonNullable.array([] as FormControl<string>[]),
+  });
+
+  get fromDate() {
+    return this.mealPlanForm.controls.dates.controls.from;
+  }
+
+  get toDate() {
+    return this.mealPlanForm.controls.dates.controls.to;
+  }
+
+  get name() {
+    return this.mealPlanForm.controls.name;
+  }
+
+  handleSuccessfulSave() {
     this.router.navigate(['recipes']);
   }
 
   constructor(
     private fb: FormBuilder,
     private router: Router,
-    public mealPlanService: MealPlanningService,
-    private http: HttpClient,
-    private route: ActivatedRoute,
-  ) {}
+    public recipeListService: RecipeListService
+  ) {
+    this.mealPlanForm.controls.dates.valueChanges.subscribe(_ => {
+      this.changedDateRangeHandler();
+      this.mealPlans = this.getDaysFromSelectedDates().map(day => ({
+        day,
+        recipes: [],
+        show: false,
+      }));
+    });
+
+    this.mealPlanForm.controls.dates.patchValue({
+      from: '2023-10-16',
+      to: '2023-10-18',
+    });
+
+    this.recipeListService.showSelectButton = true;
+  }
 
   submit() {
-    const meaPlanLoad = new FormData();
-    const form = this.mealPlanForm?.getRawValue();
-    if (!form) {
+    if (!this.mealPlanForm || this.mealPlanForm.invalid) {
+      return;
+    }
+
+    if (!this.fromDate.value || !this.toDate.value) {
       return;
     }
```

Comment:
Moving it causes the values checked here to be treated as not checked - potentially undefined - which adds complexity because we need to assure the compiler that it was already checked.